pkg/apis/core/v1alpha1/helper/helper.go

```diff
@@ -659,18 +658,6 @@ func GetShootMachineImageFromLatestMachineImageVersion(image gardencorev1alpha1.
 	return latestSemVerVersion, gardencorev1alpha1.ShootMachineImage{Name: image.Name, Version: &latestImage.Version}, nil
 }
 
-// UpdateMachineImages updates the machine images in place.
-func UpdateMachineImages(workers []gardencorev1alpha1.Worker, machineImages []*gardencorev1alpha1.ShootMachineImage) {
-	for _, machineImage := range machineImages {
-		for idx, worker := range workers {
-			if worker.Machine.Image != nil && machineImage.Name == worker.Machine.Image.Name {
-				logger.Logger.Infof("Updating worker images of worker '%s' from version %s to version %s", worker.Name, *worker.Machine.Image.Version, *machineImage.Version)
-				workers[idx].Machine.Image = machineImage
-			}
-		}
-	}
-}
-
```

Comment:
Why was this function removed - I assume it showed up as dead code? But then, the same seems to be true for the same function in `pkg/apis/core/v1beta1/helper/helper.go`?

Reply:
Yep, this was unused (usage was removed in #4438). I simply removed it here because it used the global `logger.Logger`.
I deleted the one in `pkg/apis/core/v1beta1/helper/helper.go` now as well.